packages/apputils-extension/src/themesplugins.ts

```diff
@@ -123,10 +129,58 @@ export const themesPlugin: JupyterFrontEndPlugin<IThemeManager> = {
         if (theme === manager.theme) {
           return;
         }
+        // Disable adaptive theme if users decide to change the theme when adaptive theme is on
+        if (manager.isToggledAdaptiveTheme()) {
+          return manager.toggleAdaptiveTheme();
+        }
         return manager.setTheme(theme);
       }
     });
 
+    commands.addCommand(CommandIDs.changePreferredLightTheme, {
+      label: args => {
+        const theme = args['theme'] as string;
+        const displayName = manager.getDisplayName(theme);
+        return args['isPalette']
+          ? trans.__('Use Theme: %1', displayName)
+          : displayName;
+      },
+      isToggled: args => args['theme'] === manager.preferredLightTheme,
+      execute: args => {
+        const theme = args['theme'] as string;
+        if (theme === manager.preferredLightTheme) {
+          return;
+        }
+        return manager.setPreferredLightTheme(theme);
+      }
+    });
+
+    commands.addCommand(CommandIDs.changePreferredDarkTheme, {
+      label: args => {
+        const theme = args['theme'] as string;
+        const displayName = manager.getDisplayName(theme);
+        return args['isPalette']
+          ? trans.__('Use Theme: %1', displayName)
+          : displayName;
+      },
+      isToggled: args => args['theme'] === manager.preferredDarkTheme,
+      execute: args => {
+        const theme = args['theme'] as string;
+        if (theme === manager.preferredDarkTheme) {
+          return;
+        }
+        return manager.setPreferredDarkTheme(theme);
+      }
+    });
+
+    commands.addCommand(CommandIDs.toggleAdaptiveTheme, {
+      label: trans.__('Synchronize with System Settings'),
```

Comment:
I would be more specific here too as this is visible in the palette:

```suggestion
      label: trans.__('Synchronize Color Scheme with System Settings'),
```

Reply:
I would suggest that the loner label is only used if `args['isPalette']` and the current one stays for menu to avoid the menu size exploding.